From: Jason Gerecke <killertofu@gmail.com>
Date: Tue, 1 Oct 2024 13:41:22 -0700
Subject: wacom: Do not crash when handling an unknown stylus ID

If the user brings a stylus into prox over the window which does not
have an ID known to libwacom, the program will crash due to a NULL
dereference here by `libwacom_stylus_is_erase`. The caller already
knows how to handle a NULL, so simply pass the NULL along if libwacom
was unable to find a stylus.

Closes: https://gitlab.gnome.org/GNOME/gnome-control-center/-/issues/3211
(cherry picked from commit 47d3bf94b27dabe45455d9bdb5aec9a6eeb67b61)
(cherry picked from commit bef89cb41086ad0cc522eb4832138aa3ae322f35)
---
 panels/wacom/cc-wacom-tool.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/panels/wacom/cc-wacom-tool.c b/panels/wacom/cc-wacom-tool.c
index bee0180..939b53e 100644
--- a/panels/wacom/cc-wacom-tool.c
+++ b/panels/wacom/cc-wacom-tool.c
@@ -157,6 +157,9 @@ find_stylus_for_id (WacomDeviceDatabase  *wacom_db,
 	gint i;
 
 	eraser = libwacom_stylus_get_for_id (wacom_db, tool->id);
+	if (!eraser)
+		return NULL;
+
 	if (!libwacom_stylus_is_eraser (eraser) ||
 	    libwacom_stylus_get_eraser_type (eraser) == WACOM_ERASER_BUTTON)
 		return eraser;
